src/components/Checkbox.tsx

```diff
@@ -75,21 +22,64 @@ const Checkbox: React.FC<IProps> = ({
       callback()
     }
   }
+
+  const checkboxContainerAttributes = {
+    display: "inline-block",
+    verticalAlign: "middle",
+    className: className,
+    onClick: handleClick,
+  }
+
+  const hiddenCheckboxAttributes = {
+    type: "checkbox",
+    checked: checked,
+    readOnly: true,
+    ...rest,
+  }
+
+  const styledCheckboxAttributes = {
+    "aria-hidden": true,
+    className: "styled-checkbox",
+    display: "inline-block",
+    w: `${size}rem`,
+    h: `${size}rem`,
+    minW: `${size}rem`,
+    bg: `${checked ? "primary400" : "background"}`,
+    border: "1px",
+    borderStyle: "solid",
+    borderColor: "black50",
+    borderRadius: "3px",
+    transition: "all 150ms",
+    _hover: {
+      boxShadow: "tableItemBoxShadow",
+      border: "1px",
+      borderStyle: "solid",
+      borderColor: "primary600",
+      transition: "transform 0.1s",
+      transform: "scale(1.02)",
+    },
+  }
+
   return (
-    <CheckboxContainer className={className} onClick={handleClick}>
-      <HiddenCheckbox type="checkbox" checked={checked} readOnly {...rest} />
-      <StyledCheckbox
-        aria-hidden="true"
-        checked={checked}
-        className="styled-checkbox"
-        size={size}
-      >
-        <Icon checked={checked} viewBox="0 0 24 24">
+    <Box {...checkboxContainerAttributes}>
```

Comment:
For consistency matters across the codebase, do not wrap the styles in an object. Lets just declare the props in the tag itself.

Reply:
In the last commit I removed style objects and declared the props as recommended.